File: android/src/test/java/org/ligi/passandroid/unittest/TheAppleStylePassTranslation.java
package org.ligi.passandroid.unittest;

import org.junit.Test;
import org.ligi.passandroid.model.AppleStylePassTranslation;

import static org.assertj.core.api.Assertions.assertThat;

public class TheAppleStylePassTranslation {

    @Test
    public void testThatNullTranslationWorks() {
        final AppleStylePassTranslation tested = new AppleStylePassTranslation();
        tested.loadFromString("");
        assertThat(tested.translate(null)).isNull();
    }

    @Test
    public void testThatBasicParsingWorks() {
        final AppleStylePassTranslation tested = new AppleStylePassTranslation();
        tested.loadFromString("\"foo\"=\"bar\";");
        assertThat(tested.translate("foo")).isEqualTo("bar");
    }
}

File: android/src/test/java/org/ligi/passandroid/unittest/TheAppleStylePassTranslation.kt
package org.ligi.passandroid.unittest

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.ligi.passandroid.model.AppleStylePassTranslation

class TheAppleStylePassTranslation {

    @Test
    fun testThatNullTranslationWorks() {
        val tested = AppleStylePassTranslation()
        tested.loadFromString("")
        assertThat(tested.translate(null)).isNull()
    }

    @Test
    fun testThatBasicParsingWorks() {
        val tested = AppleStylePassTranslation()
        tested.loadFromString("\"foo\"=\"bar\";")
        assertThat(tested.translate("foo")).isEqualTo("bar")
    }
}
